File: service/api/validate.go
package api

import (
	"fmt"
	"strconv"
)

func validateBinary(binary string) error {
	b, err := strconv.ParseInt(binary, 2, 0)
	if err != nil || b < 1 {
		return fmt.Errorf("invalid input: is not a binary number or greater than 1: %s", binary)
	}
	return nil
}

func validatePositiveDecimal(decimal int) error {
	if decimal < 1 {
		return fmt.Errorf("invalid input: is not a decimal number or greater than 1: : %d", decimal)
	}
	return nil
}

func validateHexadecimal(hexadecimal string) error {
	h, err := strconv.ParseInt(hexadecimal, 16, 0)
	if err != nil || h < 1 {
		return fmt.Errorf("invalid input: is not a hexadecimal number or greater than 1: %s", hexadecimal)
	}
	return nil
}

File: public/scripts/src/bin/validate.ts
export const isValidBinary = (binary: string): boolean => {
  if (binary.length > 64) {
    return false;
  }

  for (let i = 0; i < binary.length; i++) {
    if (binary[i] !== "0" && binary[i] !== "1") {
      return false;
    }
  }
  return true;
};

export const isValidDecimal = (decimal: number): boolean => {
  return !isNaN(decimal) && decimal < 999999999999;
};

export const isValidHexadecimal = (hexadecimal: string): boolean => {
  return /^[A-F0-9]+$/.test(hexadecimal) && hexadecimal.length < 64;
};
